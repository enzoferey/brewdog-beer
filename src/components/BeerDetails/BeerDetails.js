import React from "react";
import PropTypes from "prop-types";
import style from "./beer-details.scss";

import PageContainer from "components/PageContainer";

const BeerDetails = ({ image, name, abv, description }) => (
  <PageContainer>
    <div className={style.details}>
      <img src={image} alt="" />
      <h2>{name}</h2>
      <h4>{`${abv}%`}</h4>
      <p>{description}</p>
    </div>
  </PageContainer>
);

BeerDetails.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  abv: PropTypes.number,
  description: PropTypes.string,
};

export default BeerDetails;
