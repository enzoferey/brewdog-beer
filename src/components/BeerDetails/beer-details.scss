.details {
  text-align: center;
}

.photo {
  height: 308px;
  margin-bottom: 24px;
}

.name {
  font-size: 55px;
  margin: 0 0 11px;
}

.attributes {
  font-size: 24px;
  color: #333;
  font-weight: 400;
}

.description {
  font-size: 20px;
  margin-top: 45px;
  text-align: left;
}
