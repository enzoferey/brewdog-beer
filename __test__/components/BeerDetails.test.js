import React from "react";
import BeerDetails from "components/BeerDetails";

const props = {
  image: "src-path",
  name: "Beer name",
  abv: 5,
  description: "Lorem ipmsum",
};

test("renders correctly", () => {
  testSnapshot(<BeerDetails {...props} />);
});
