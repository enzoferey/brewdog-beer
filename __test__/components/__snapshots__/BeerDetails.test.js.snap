// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <img
    alt="Beer bottle image"
    src="src-path"
  />
  <h2>
    Beer name
  </h2>
  <h4>
    Alc. 5% By Vol.
  </h4>
  <p>
    Lorem ipmsum
  </p>
</div>
`;
